src/java/org/apache/cassandra/index/sai/disk/hnsw/CassandraOnHeapHnsw.java

```diff
@@ -54,11 +55,66 @@
 import org.apache.lucene.util.StringHelper;
 import org.apache.lucene.util.hnsw.HnswGraphSearcher;
 import org.apache.lucene.util.hnsw.NeighborQueue;
+import org.apache.lucene.util.hnsw.RandomAccessVectorValues;
 
 public class CassandraOnHeapHnsw<T>
 {
     private static final Logger logger = LoggerFactory.getLogger(CassandraOnHeapHnsw.class);
 
+    private class ReorderingRamAwareVectorValues implements RamAwareVectorValues {
+        private final RamAwareVectorValues vectors;
+        private final BiMap <Integer, Integer> ordinalMap;
+
+        public ReorderingRamAwareVectorValues (RamAwareVectorValues vectors, BiMap <Integer, Integer> ordinalMap) {
+            this.vectors = vectors;
+            this.ordinalMap = ordinalMap;
+        }
+
+        @Override
+        public long write(SequentialWriter writer) throws IOException
+        {
+            return vectors.write(writer);
```

Comment:
`vectors` will be either `ConcurrentVectorValues` or `CompactionVectorValues`. If we are invoking `write()` for either of these classes, do we actually end up using `ReorderingRamAwareVectorValues.ordinalMap`?

Reply:
the reordering here happens in vectorValue

Reply:
Yes, I see that `ReorderingRamAwareVectorValues.vectorValue()` applies the permutation map prior to leveraging `{Concurrent,Compaction}.vectorValue()`.

However, `ReorderingRamAwareVectorValues.write()` :
- defers immediately to the wrapped `{Concurrent,Compaction}VectorValues` object's `write` method
- I'm not immediately seeing either of those classes' `write` implementations get any original ordinal (and corresponding vector) before serializing to disk

I may be missing something very silly here.

Reply:
I changed it to a simpler thing